libsolidity/parsing/Token.h

```diff
@@ -403,5 +311,29 @@ class Token
 	static char const m_tokenType[NUM_TOKENS];
 };
 
+class ElementaryTypeNameToken
+{
+public:
+	ElementaryTypeNameToken(Token::Value token, std::string const& description)
+	{
+		solAssert(isElementaryTypeName(token, description), "");
+		std::tie(m_name, M, N) = setTypes(token, description);
+		tok = token;
+	}
+	
+	std::string toString(bool const& tokValue = false) const& { return tokValue ? Token::toString(tok) : m_name; }
+	unsigned int const& returnM() const& { return M; }
```

Comment:
Please do not use `return` as a prefix for getters. Getters should be named exactly as the members, just without `m_` prefix. And perhaps replace `M` and `N` by `firstNumber` and `secondNumber` (only here, not in the token names)?